common/nymsphinx/forwarding/src/packet.rs

```diff
@@ -174,13 +179,51 @@ impl MixPacket {
         })
     }
 
-    pub fn into_v2_bytes(self) -> Result<Vec<u8>, MixPacketFormattingError> {
+    pub fn into_v2_bytes(&self) -> Result<Vec<u8>, MixPacketFormattingError> {
```

Comment:
why do we need to preserve the old packet?
if there's a reason for it, the new method should be exposed as `to_v2_bytes`. by convention an `into` consumes `self`

Reply:
I can change the method to follow the convention